src/csv/lexer.cr

```diff
@@ -89,15 +89,15 @@ abstract class CSV::Lexer
       when '\0'
         raise "unclosed quote"
         break
-      when '"'
+      when @quote_char
         case next_char
-        when ','
+        when @separator
           check_last_empty_column
           break
         when '\r', '\n', '\0'
           break
-        when '"'
-          @buffer << '"'
+        when @quote_char
```

Comment:
Thank you @asterite, I did understand the reason behind the recommendation of the instance variable usage, but was wondering if it applies to *this* `case` case (no pun intended):

```crystal
when @quote_char
  @buffer << @quote_char
```

It is safe to assume, given that context that: `@quote_char` usage is safe, since is set during initialization, the variable is not reassigned and has no setter from the outside that could change it.